core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -166,7 +166,12 @@ public synchronized void shutdown() {
 
   @Override
   public String toString() {
-    return super.toString() + "(" + name + ")";
+    return getLogId() + "(" + name + ")";
+  }
+
+  @Override
+  public String getLogId() {
+    return "InProcessTransport@" + Integer.toHexString(hashCode());
```

Comment:
FYI, hashcodes will be reused within the same process as transports come and go.